model/src/pyrenew/process/simplerandomwalk.py

```diff
@@ -77,16 +77,16 @@ def sample(
         def transition(x_prev, _):
             # numpydoc ignore=GL08
             diff, *_ = self.step_rv(**kwargs)
-            x_curr = x_prev + diff
+            x_curr = x_prev + diff.value
             return x_curr, x_curr
 
         _, x = scan(
             transition,
-            init=init,
+            init=init.value,
             xs=jnp.arange(n_steps - 1),
         )
 
-        return (jnp.hstack([init, x.flatten()]),)
+        return (SampledValue(jnp.hstack([init.value, x.flatten()])),)
```

Comment:
As above, I think we should probably pass `self.t_start` and `self.t_unit` here, even if they're sometimes going to be `None`.

Reply:
Noted